src/macros.jl

```diff
@@ -1005,34 +1005,128 @@ function parsevariable(_error::Function, infoexpr::VariableInfoExpr, lvalue, lsi
-#   `buildvariable` is called separately for each variable of the container. Moreover they will be passed inside the VariableInfo object.
-# * A custom error function is passed as positional argument to print the full @variable call before the error message.
-# * The keyword argument basename is not passed to `constructconstraint!` but rather to `addvariable`.
-#
-# Examples:
-# * `@variable(m, x >= 0)` is equivalent to `x = addvariable(m, buildvariable(msg -> error("In @variable(m, x >= 0): ", msg), VariableInfo(true, 0, false, NaN, false, NaN, false, NaN, false, false)), "x")`
-# * `@variable(m, x[1:N,1:N], Symmetric, Poly(X))` is equivalent to
-#   ```
-#   x = Matrix{...}(N, N)
-#   for i in 1:N
-#       for j in 1:N
-#           x[i,j] = x[j,i] = addvariable(m, buildvariable(msg -> error("In @variable(m, x[1:N,1:N], Symmetric, Poly(X)): ", msg), VariableInfo(false, NaN, false, NaN, false, NaN, false, NaN, false, false), Poly(X)), "")
-#       end
-#   end
-#   ```
+"""
+    @variable(m; kwargs...)
+
+Add an *anonymous* (see [Names](@ref)) variable to the model `m`
+described by the keyword arguments `kwargs`.
+
+    @variable(m, expr, args...; kwargs...)
+
+Add a variable to the model `m` described by the expression `expr`, the
+positional arguments `args` and the keyword arguments `kwargs`. The expression
+`expr` can either be (note that in the following the symbol `<=` can be used
+instead of `≤` and the symbol `>=`can be used instead of `≥`)
+
+* of the form `varexpr` creating variables described by `varexpr`;
+* of the form `varexpr ≤ ub` (resp. `varexpr ≥ lb`) creating variables described by
+  `varexpr` with upper bounds given by `ub` (resp. lower bounds given by `lb`);
+* of the form `varexpr == value` creating variables described by `varexpr` with
+  fixed values given by `value`; or
+* of the form `lb ≤ varexpr ≤ ub` or `ub ≥ varexpr ≥ lb` creating variables
+  described by `varexpr` with lower bounds given by `lb` and upper bounds given
+  by `ub`.
+
+The expression `varexpr` can either be
+
+* of the form `varname` creating a scalar real variable of name `varname`;
+* of the form `varname[...]` or `[...]` creating a container of variables (see
+  [Containers in macro](@ref).
+
+The recognized positional arguments in `args` are the following:
+
+* `Bin`: Sets the variable to be binary, i.e. either 0 or 1.
+* `Int`: Sets the variable to be integer, i.e. one of ..., -2, -1, 0, 1, 2, ...
+* `Symmetric`: Only available when creating a square matrix of variables, i.e.
+  when `varexpr` is of the form `varname[1:n,1:n]` or `varname[i=1:n,j=1:n]`.
+  It creates a symmetric matrix of variable, that is, it only creates a
+  new variable for `varname[i,j]` with `i ≤ j` and sets `varname[j,i]` to the
+  same variable as `varname[i,j]`.
+* `PSD`: Same as `Symmetric` but also constrains the matrix to be positive
```

Comment:
Nit: "Same as Symmetric" is confusing. You can say that it is both `Symmetric` and constrained to be PSD.